Feauture/signup

Aya921 opened the pull request:
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

✅ Added **Signup** feature:

- Built the Signup screen UI using Flutter widgets.
- Integrated Signup API using Retrofit & Dio.
- Followed Clean Architecture: separated logic into `data`, `domain`, and `presentation` layers.
- Used Cubit for managing signup state and handling success/error scenarios.
- Saved authentication token securely using SharedPreferences after successful registration.

## Related Tickets & Documents

<!-- Add related issue IDs if available -->
<!-- Example: Closes #45 -->

## QA Instructions, Screenshots, Recordings

To test this feature:

1. Go to the Signup screen.
2. Fill in name, email, and password:
   - Try valid data → expect success and navigation.
   - Try invalid data → expect proper validation messages.
3. After successful signup:
   - Confirm token is saved correctly in SharedPreferences.
   - (Optional) Test token usage in future requests.

**Screenshot:**
![Signup UI](https://github.com/user-attachments/assets/4369f8ab-c1de-413f-9758-d053eb1b02aa)


### UI accessibility checklist

- [x] Used semantic Flutter widgets (TextField, ElevatedButton, etc.)
- [x] Form fields validated and keyboard accessible
- [x] Checked contrast and error states manually

## Added/updated tests?

- [ ] Yes
- [x] No, _(will be added later / not in current scope)_

## [optional] Are there any post deployment tasks we need to perform?

- None. The token is automatically stored via SharedPreferences on successful signup.

## [optional] What gif best describes this PR or how it makes you feel?

![All set!](https://media.giphy.com/media/111ebonMs90YLu/giphy.gif)

Aya921 committed:
feat: Implement user signup functionality with validation and localization

- Added localization strings for signup process in English and Arabic.
- Created TokenStorage interface and its implementation using SharedPreferences.
- Developed API services for user signup, including request and response models.
- Implemented user entity and repository for managing user data.
- Added signup use case to handle business logic for user registration.
- Created view model for managing signup state and validation.
- Developed signup page UI with form validation and user feedback.
- Updated routing to navigate to the signup page.
- Enhanced error handling for API responses during signup.

Aya921 committed:
feat: Refactor token management by introducing TokenService and removing TokenStorage

ali72-20 commented on lib/core/constant/endPoints_constants/endpoints.dart:
```diff
@@ -3,7 +3,8 @@ class Endpoints {
 
```
make this class `abstract` or with 
`private constructor `to block any one can create instance from this class which is no meaning for this

ali72-20 commented on lib/core/services/token_service.dart:
```diff
@@ -0,0 +1,35 @@
+@injectable
+class TokenService {
+
+  final SharedPreferences _prefs;
```
token is very sensitive data for user and shared pref create a directry on android files and any one can find this token and get all user data 

so we need to hash or secure this token then you need to search for` flutter secure storage package `

ali72-20 commented on lib/core/services/token_service.dart:
```diff
@@ -0,0 +1,35 @@
+
+  
+  Future<String?> getToken()async {
+    return   _prefs.getString(Endpoints.token);
```
token is not endpoint it's a const key word

ali72-20 reviewed (commented):
great work for beginning 
next time will be more better

Aya921 committed:
feat: Update localization keys for user input validation messages

- Renamed error keys in app_en.arb and app_ar.arb to include "empty" prefix for clarity.
- Added new validation messages for username, first name, last name, email, password, confirm password, and phone number.
- Updated AppLocalizations class to reflect new keys and added corresponding translations in Arabic.
- Refactored token service to use FlutterSecureStorage instead of SharedPreferences for enhanced security.
- Updated signup response models to use constants for JSON keys for better maintainability.
- Enhanced signup validators to utilize localized error messages.
- Implemented Bloc pattern for signup events and states to manage signup process more effectively.
- Updated signup page to handle validation and user signup events using the new Bloc structure.
- Registered necessary plugins for secure storage on Linux and Windows platforms.

Aya921 committed:
feat: Implement user authentication flow with sign-in functionality

- Added Arabic and English translations for login and sign-up screens.
- Created login and forget password pages with corresponding routes.
- Implemented API services for user sign-in, including request and response models.
- Developed SigninUsecase and AuthRepo for handling sign-in logic.
- Integrated SigninCubit for managing sign-in state with Bloc pattern.
- Built UI components for login form and footer, including validation and loading states.
- Updated main app route to start with the login page.
- Added connectivity_plus package for network connectivity checks.

Aya921 committed:
feat: Enhance dependency injection and clean up authentication flow
